ast/src/parsed/utils.rs

```diff
@@ -50,8 +51,10 @@ where
         | PilStatement::PolynomialDefinition(_, _, e)
         | PilStatement::PolynomialIdentity(_, e)
         | PilStatement::PublicDeclaration(_, _, _, e)
-        | PilStatement::ConstantDefinition(_, _, e)
-        | PilStatement::LetStatement(_, _, Some(e)) => postvisit_expression_mut(e, f),
+        | PilStatement::ConstantDefinition(_, _, e) => postvisit_expression_mut(e, f),
+        // TODO we are not visiting the expressions inside the type name here.
```

Comment:
still need to fix this

Reply:
I'm tempted to implement @Schaeff 's version of the visitor before finishing this pr...